### PR TITLE
AWS improvements

### PR DESCRIPTION
**Pull request by prestwich**
Supersedes #170

## Motivation

Improve ops ergonomics wrt AWS services

## Solution

- changes S3 & KMS to allow more standard credentials types
- read region from the environment
- remove region from KMS config specification

## PR Checklist

- [ ] Added Tests
- [x] Updated Documentation
- [x] Updated CHANGELOG.md for the appropriate package

### CODE REVIEW

**Comment by prestwich** [comment]
@arnaud036 this now includes an integration test for the KMS that runs only if `TEST_KMS_ID` is set. The integration test attempts to sign a message, using the env-provided region, and the key specified by the `TEST_KMS_ID` env var

File: nomad-core/tests/signer_integration.rs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+use ethers_signers::Signer;
+use nomad_core::Signers;
+use nomad_xyz_configuration::{agent::SignerConf, FromEnv};
+
+#[tokio::test]
+async fn signer_auths() {
+    if let Some(signer_conf) = SignerConf::from_env("TEST_KMS") {
```
Comment (ltchang2019) [inline nomad-core/tests/signer_integration.rs:7]: Changes to `from_env` here: https://github.com/nomad-xyz/rust/pull/171

Comment (prestwich) [reply nomad-core/tests/signer_integration.rs:7]: ooh, can we fix that to take a `&[&str]` with prefixes in precedence order?

File: configuration/src/agent/signer.rs [hunk 2]
```diff
@@ -12,10 +40,12 @@ pub enum SignerConf {
...
-        /// The UUID identifying the AWS KMS Key
-        id: String, // change to no _ so we can set by env
-        /// The AWS region
-        region: String,
+        /// An AWS identifier for the key. This may be
+        /// 1. Its UUID
+        /// 2. Its ARN
+        /// 3. A key alias
+        /// 4. A key alias's ARN
+        id: String,
```
Comment (ltchang2019) [inline configuration/src/agent/signer.rs:48]: Was not aware you could pass in different types of IDs when instantiating AWS signer. Could you comment in reference to documentation that describes this behavior?

File: agents/processor/src/push.rs [hunk 3]
```diff
@@ -41,12 +39,9 @@ impl std::fmt::Debug for Pusher {
...
     /// Instantiate a new pusher with a region
-    pub fn new(name: &str, bucket: &str, region: Region, db: NomadDB) -> Self {
-        let client = S3Client::new_with(
-            HttpClient::new().unwrap(),
-            EnvironmentProvider::with_prefix(AWS_S3_PREFIX),
-            region.clone(),
-        );
+    pub fn new(name: &str, bucket: &str, db: NomadDB) -> Self {
+        let region: Region = Default::default(); // loads from aws env
+        let client = S3Client::new(region.clone());
```
Comment (ltchang2019) [inline agents/processor/src/push.rs:44]: Did you run this by @arnaud036? Would need to add env var

**Review by ltchang2019 — COMMENTED** [review]
Can we remove the `..._REGION` fields from the test fixtures now that they're irrelevant? 

Overall though looks good as long as:
- @arnaud036 is good with env var changes
- we can successfully run in dev

File: nomad-core/tests/signer_integration.rs (hunk 1 above)
Comment (ltchang2019) [reply nomad-core/tests/signer_integration.rs]: Replacing/simplifying with this interface https://github.com/nomad-xyz/rust/pull/180

File: agents/processor/src/push.rs (hunk 3 above)
Comment (prestwich) [reply agents/processor/src/push.rs:44]: yep! we're trying to move towards more industry-standard AWS configuration. this is part of that :)

**Review by ltchang2019 — APPROVED** [review]

**Pull request merged**